### ISSUE TITLE
PRD: Job card score intelligence — dream rank badge + score breakdown + Score label

### ISSUE THREAD

**Issue by narendranathe** [narendranathe/job-scout#114]
## Overview

Job cards currently show a bare relevance percentage with no context for what drove it and no signal about how important the company is to the user personally. This PRD adds three meaningful signals to every card: a **dream company rank badge** (where this company sits in the user's personally ordered list), a **score breakdown** (which components contributed), and a **Score label** above the chip. The sequential position badge added in #113 is removed — position in current search results has no durable meaning.

## Problem Statement

The relevance score chip shows e.g. `74%` but gives no clue whether that score is driven by strong skill alignment, a platinum company boost, or title match alone. A 74% from Anthropic (dream company #1) feels different from a 74% at a random company — but the card looks identical. Users must mentally reconstruct why a job scored well instead of seeing it at a glance.

## Detailed Description

**1. Remove sequential position rank (#113 rollback)**
Remove the `rank` prop from `JobCard.jsx` and `rank={idx+1}` from `JobsTab.jsx`. Position in the current search result set changes with every filter and has no intrinsic value.

**2. Add "Score" label above the score chip**
The score chip currently shows `74%` with no label. Add a small `Score` text label so its meaning is self-describing. Keep the chip color scale unchanged.

**3. Dream company rank badge (card meta row)**
Each of the user's 41 dream companies has a personal rank based on the ordered `_dream_companies_reference` list in `profile.py`:

Anthropic=1, OpenAI=2, Stripe=3, Databricks=4, Snowflake=5, Goldman Sachs=6, Walmart=7, Apple=8, NVIDIA=9, Google=10, Microsoft=11, Disney=12, Citadel=13, Citadel Securities=14, AQR Capital=15, Hudson River Trading=16, Jane Street=17, Two Sigma=18, Jump Trading=19, SIG=20, IMC Trading=21, Bridgewater Associates=22, Flow Traders=23, Tower Research Capital=24, Millennium Management=25, Point72=26, Optiver=27, Virtu Financial=28, PIMCO=29, Netflix=30, Meta=31, Spotify=32, Fidelity=33, Uber=34, Bloomberg=35, Morgan Stanley=36, BlackRock=37, DoorDash=38, Amazon=39, Salesforce=40, JP Morgan Chase=41

Show a `★ Dream #N` chip in the card meta row for companies on this list. Companies not on the list show nothing. Decorative only, no click interaction.

**4. Score breakdown in expanded card body**
When the user expands a card, show an "Estimated score breakdown" panel above the skills pills. Each scoring component displayed as a labeled mini-bar row showing label, filled bar, estimated %, and driver text (matched skill names or keyword found):

- Core skills (max 36%): N/15 matched, skill names listed
- Secondary skills (max 17%): N/35 matched
- Title relevance (max 15%): keyword tier matched
- Location (max 10%): remote / TX / other
- Experience level (max 10%): senior/staff/lead detected or not
- Sponsorship (+/-8%): H1B signal direction
- Salary tier (max 4%): salary_max bracket
- Platinum boost (+8%): Tier 0 company or not

Footer shows estimated total vs stored score for transparency.

**Frontend-only implementation:** `matched_skills` in the job payload already contains all matched skills. Split into core/secondary by checking against `CORE_SKILLS`/`SECONDARY_SKILLS` lists added to `jobConstants.js`. No backend changes needed.

## User Stories

- As a job seeker, I want to see "Dream #3" on a Databricks card so I immediately know this is a priority company
- As a job seeker, I want to see how the 74% score breaks down so I understand whether it is skill-aligned or inflated by boosters
- As a job seeker, I want the score chip labeled "Score" so it is unambiguous
- As a job seeker, I do not want a search-result position number on my cards because it changes with every filter

## Acceptance Criteria

- [ ] Sequential rank badge removed from all job cards (JobCard + JobsTab)
- [ ] "Score" label renders above the score percentage chip
- [ ] Dream rank badge renders in card meta row for all 41 dream companies
- [ ] Dream rank badge does NOT render for non-dream companies
- [ ] Score breakdown panel renders in expanded card body (not collapsed header)
- [ ] Breakdown shows 8 rows: Core skills, Secondary skills, Title, Location, Experience, Sponsorship, Salary, Platinum
- [ ] Each row shows: label, filled mini-bar, estimated % value, driver text
- [ ] Breakdown total within +-2% of displayed score chip value
- [ ] Breakdown labeled "Estimated score breakdown"
- [ ] No regression: % on score chip, existing expanded content intact

## Non-Functional Requirements

- Performance: `estimateScoreBreakdown()` is O(n) in matched_skills length; zero new network calls
- Accuracy: Breakdown total within +-2% of stored `relevance_score`

### Score weights (from `backend/core/relevance.py`)

```
CORE_WEIGHT        = 0.36  (/ 15 core skills = 0.024 per skill)
SECONDARY_WEIGHT   = 0.17  (/ 35 secondary skills = 0.00486 per skill)
TITLE_WEIGHT       = 0.15  (tiered: 15% data eng, 14% ml eng, 13% quant, 12% analytics eng, 8% platform, 5% fallback)
LOCATION_WEIGHT    = 0.10
EXPERIENCE_WEIGHT  = 0.10
SPONSORSHIP_WEIGHT = 0.08  (+ or -)
SALARY_WEIGHT      = 0.04
PLATINUM_BOOST     = 0.08  (additive after clamp, tier=0 companies)
```

### Core skills list (15, from profile.py)
python, sql, spark, pyspark, kafka, airflow, etl, data pipeline, data lake, data warehouse, azure, aws, databricks, delta lake, microsoft fabric

### Secondary skills list (35, from profile.py)
docker, kubernetes, terraform, ci/cd, dbt, snowflake, redshift, bigquery, postgresql, sql server, mongodb, fastapi, flask, rest api, git, linux, bash, mlflow, mlops, machine learning, pytorch, tensorflow, scikit-learn, tableau, powerbi, looker, flink, kinesis, data mesh, numpy, pandas, scipy, quantitative, statistical modeling, backtesting

### Existing Code Affected
- `frontend/src/lib/jobConstants.js`: add CORE_SKILLS, SECONDARY_SKILLS, DREAM_COMPANY_RANK, getDreamRank(), estimateScoreBreakdown()
- `frontend/src/components/JobCard.jsx`: remove rank prop/badge; add Score label + dream rank chip + breakdown panel
- `frontend/src/tabs/JobsTab.jsx`: remove rank={idx+1}, revert map lambda

### Module 1: jobConstants.js — data + helpers (M)
- **Responsibility:** Add skill lists, dream rank map, and two pure helpers
- **Interface:** CORE_SKILLS[], SECONDARY_SKILLS[], DREAM_COMPANY_RANK{string->number}, getDreamRank(company)->number|null, estimateScoreBreakdown(job)->{label,maxPct,estPct,barFill,driver}[]
- **Dependencies:** None (pure data)
- **Key files:** frontend/src/lib/jobConstants.js

### Module 2: JobCard.jsx — UI (M)
- **Responsibility:** Remove rank badge; add Score label; render dream rank chip in meta row; render breakdown panel above skills pills in expanded body
- **Interface:** Props: remove rank. Internal calls to getDreamRank + estimateScoreBreakdown.
- **Dependencies:** Module 1
- **Key files:** frontend/src/components/JobCard.jsx

### Module 3: JobsTab.jsx — cleanup (S)
- **Responsibility:** Remove rank={idx+1} from JobCard call; revert .map((j,idx) to .map(j
- **Dependencies:** None
- **Key files:** frontend/src/tabs/JobsTab.jsx

## Dependency Graph

```
Module 1 (jobConstants.js)
    +-- Module 2 (JobCard.jsx)

Module 3 (JobsTab.jsx)  [independent, parallel]
```

## Out of Scope
- User-editable dream company ranking UI
- Backend score breakdown export
- Score breakdown in collapsed card header
- RareTab card score breakdown
- Color tiers for dream badge (gold/silver/bronze)

## Definition of Done
- [ ] All acceptance criteria met
- [ ] Vite build clean
- [ ] Anthropic card shows Dream #1 + breakdown; non-dream card has no badge; total within +-2%
- [ ] No regressions in other tabs
